### PR TITLE
[vernac] Remove obsolete Existential and Grab Existential variables.

### PR DESCRIPTION
**Pull request by ejgallego**
This is a proposal to eventually remove the commands `Existential` and
`Grab Existential`. To my understanding these commands are supposed to
be deprecated and scheduled to removal, but not 100% sure so we would
need to confirm.

This would of course need a deprecation phase, let's see the impact on
CI first.

### CODE REVIEW

**Comment by herbelin** [comment]
For `Grab Existentials`, my memories are that there are indeed cases where `Unshelve` misses some evars, as a consequence of cases where an evar fails to be correctly tracked in the shelf. I vaguely remember that @mattam82 reported a case once (but I don't know how to find it again). So, I guess the idea would indeed be to first ensure that all evars are tracked in either the shelf, the given_up list or the main list (@aspiwack, @ppedrot, is it correctly the idea?).

**Comment by Zimmi48** [comment]
I wanted to do this too but the problem was indeed that we have a few bugs around that make it necessary to rely on `Grab Existential Variables`. This PR should be a good way of finding the remaining uses in external projects and whether `Unshelve` can be used instead.

**Comment by aspiwack** [comment]
@herbelin it sounds correct, indeed.

**Comment by ejgallego** [comment]
Ok, so I've tracked down a few issues that could be relevant:

https://github.com/coq/coq/issues/9171
https://github.com/coq/coq/issues/8047
https://github.com/coq/coq/issues/6202
https://github.com/coq/coq/issues/4509
https://github.com/coq/coq/issues/8770
https://github.com/coq/coq/issues/5570

File: CHANGES.md [hunk 1]
```diff
@@ -39,6 +39,9 @@ Specification language, type inference
   solved by writing an explicit `return` clause, sometimes even simply
   an explicit `return _` clause.
 
+- The commands `Existential` and `Grab Existential` have been
```
Comment (gares) [inline CHANGES.md]: The command is `Grab Existential Variables`, and `xxx` should be fixed before merging

**Review by gares — CHANGES_REQUESTED** [review]

**Comment by gares** [comment]
I'm positive on this. Also, some users asked for a sort of "automatic unshelve and eventually solve" tactic to be run when one attempts a Qed and there are still evars on the shelf. At least this is what I remember of the discussions we had, CC @CohenCyril

**Comment by gares** [comment]
I mean, it may be worth understanding is all use cases of `Unshelve` fall in this category (you unshelve at the very end and you run the same tactic).

**Comment by gares** [comment]
Yes

**Comment by ejgallego** [comment]
@ppedrot I was doing a large clenaup of proof global global etc... And I was wondering about this functions. I don't recall now if these functions pose a true problem but will update the PR based on your remarks next week.

The current or was maybe more of an experiment to see impact.

**Comment by ejgallego** [comment]
Indeed I confirm this is not needed for any of the pending projects related to document / UI / proof cleanup.

However it does seem to me that the commands are deprecated so indeed we should discuss a bit more before closing IMHO.

**Comment by ejgallego** [comment]
Closed in favor of #9545

**Pull request closed without merging**